sdk/containerregistry/azure-containerregistry/azure/containerregistry/_models.py

```diff
@@ -127,58 +146,101 @@ def _to_generated(self) -> ManifestWriteableProperties:
 
     @property
     def architecture(self) -> ArtifactArchitecture:
+        """CPU Architecture of an artifact.
+        :rtype: ~azure.containerregistry.ArtifactArchitecture
+        """
         return self._architecture
 
     @property
     def created_on(self) -> datetime:
+        """Time and date an artifact was created.
+        :rtype: ~datetime.datetime
+        """
         return self._created_on
 
     @property
     def digest(self) -> str:
+        """Digest for the artifact.
+        :rtype: str or None
```

Comment:
Doesn't this mean the return type should be `Union[str, None]`?

Reply:
Instead of `Union[str, None]` write `Optional[str]`

Reply:
This property is required in generated model, so it should be `str`. Just addressed the `:rtype:`. 
Additionally, double checked other properties and other models in this file as well.